src/jllvm/compiler/CodeGenerator.hpp

```diff
@@ -138,19 +142,22 @@ class CodeGenerator
     /// This function must be only called once. 'generatePrologue' is used to initialize the local variables and
     /// operand stack at the start of the method. 'offset' is the bytecode offset at which compilation should start and
     /// must refer to a JVM instruction.
-    void generateBody(PrologueGenFn generatePrologue, std::uint16_t offset = 0);
+    llvm::Value* generateBody(PrologueGenFn generatePrologue, std::uint16_t offset = 0);
 };
 
 /// Generates new LLVM code at the back of 'function' from the JVM Bytecode in 'method'.
 /// 'generatePrologue' is called by the function to initialize the operand stack and local variables at the beginning of
 /// the newly created code. 'offset' is the bytecode offset at which compilation should start and must refer to a JVM
 /// instruction.
-inline void compileMethodBody(llvm::Function* function, const Method& method,
-                              CodeGenerator::PrologueGenFn generatePrologue, std::uint16_t offset = 0)
+/// A basic block without a terminator is created that all return instructions branch to instead of calling return.
+/// If the method returns void, this basic block is returned. Otherwise, a PHI instruction within the basic block
+/// containing the value that should be returned is returned instead.
+inline llvm::Value* compileMethodBody(llvm::Function* function, const Method& method,
```

Comment:
Could you use something like a `variant` or `PointerUnion` here to signify the difference the between the two cases?